### PR TITLE
Convert to a package, use pytest, debug tests, and do some clean-up

### PR DESCRIPTION
**Pull request by seav**
### Proposed changes in this pull request
- Convert into a Python package:
    - Add **setup.py**.
    - Move main files and directories into the `cadasta.test` module (following the convention laid out in the cadasta-sdk repository) and add **\_\_init\_\_.py** files as needed.
    - Rename directories containing test files to **\*_test**.
    - Rename **resources** directory to **files** so that test files that are not resources can be added here.
- Use pytest:
    - Delete **cadasta_test_suites.py** because tests should be automatically discoverable instead of manually adding them to a list. Note: tests that were commented out in this file are also made undiscoverable from pytest by renaming the test "xtest\_\*" instead of "test\_\*".
    - Replace **runtests** shell script with **runtests.py** Python script.
    - Add argparse to **runtests.py** and update arguments of **runtests.py** to allow passing optional arguments to pytest and to allow specifying the Selenium WebDriver to use (Chrome by default, and Firefox as an alternative). Update **README.md** to match.
- Debug some tests:
    - With the proper test fixtures, all account tests should now pass.
    - Remove references to and tests for the second password field that no longer exists in the UI.
    - Update error messages.
    - Avoid updating the entity data.
- Do some more clean-up:
    - Make internal import paths relative instead of absolute.
    - Move the `setUp()` and `tearDown()` methods from the child classes to the parent `SeleniumTestCase` class, and add `super()` calls as needed.

### When should this PR be merged
When convenient.

### Risks
None foreseen.

### Follow-up actions
None (yet).

#### General

- [ ] **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description. 
- [ ] **Is the PR labeled correctly?** It should have the `migration` label if a new migration is added. 
- [ ] **Is the risk level assessment sufficient?** The risks section should contain all risks that might be introduced with the PR and which actions we need to take to mitigate these risks. Possible risks are database migrations, new libraries that need to be installed or changes to deployment scripts.

#### Functionality

- [ ] **Are all requirements met?** Compare implemented functionality with the requirements specification.

#### Code

- [ ] **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.

#### Tests

- [ ] **Are there sufficient test cases?** Ensure that all components are tested individually; models, forms, and serializers should be tested in isolation even if a test for a view covers these components. 

#### Security

- [ ] **Confirm this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.**
- [ ] **Does this solution handle outliers and edge cases gracefully?**
- [ ] **Are all external communications secured and restricted to SSL?**

### CODE REVIEW

File: cadasta/test/account_tests/test_registration.py [hunk 1]
```diff
@@ -26,80 +24,61 @@ def test_new_registration(self):
...
+                "'A user with that username already exists')]")
             elems.extend(self.wd.find_elements_by_xpath(
                 "//*[contains(text(), "
                 "'Another user with this email already exists')]"))
             if len(elems) == 0:
                 username_available = True
-                Credentials().set_test_username(test_username)
-                Credentials().set_test_password(test_password)
-                Credentials().set_test_email(test_username + "@example.com")
+                # Credentials().set_test_username(test_username)
```
Comment (oliverroick) [inline cadasta/test/account_tests/test_registration.py]: Is that something you plan to add back later? Otherwise, we should just remove the lines.

File: cadasta/test/resource_tests/test_location_resources.py [hunk 2]
```diff
@@ -1,18 +1,16 @@
...
+from ..pages import ProjectsPage
 
-class AddLocationResource(SeleniumTestCase):
 
-    def setUp(self):
-        self.wd = CustomWebDriver()
+class AddLocationResource(SeleniumTestCase):
 
-    def test_attach_existing_resource_to_new_location(self):
+    def xtest_attach_existing_resource_to_new_location(self):
```
Comment (oliverroick) [inline cadasta/test/resource_tests/test_location_resources.py]: Is the rename of the method a temporary measure?

File: cadasta/test/runtests.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+import os
+import sys
+import pytest
+from subprocess import Popen, DEVNULL
+
+
+if __name__ == '__main__':
+    # cadasta-test does not require Django
+    if 'DJANGO_SETTINGS_MODULE' in os.environ:
+        del os.environ['DJANGO_SETTINGS_MODULE']
```
Comment (oliverroick) [inline cadasta/test/runtests.py]: I don't understand why you are removing the env variable. Can you clarify?

File: cadasta/test/account_tests/test_registration.py (hunk 1 above)
Comment (seav) [inline cadasta/test/account_tests/test_registration.py]: Some tests require this behavior but this messes up subsequent tests. This is because with pytest, some class state is maintained between tests. I just commented these lines out while I haven't decided yet what to do replace these with, if ever I decide to replace them while debugging the other tests.

File: cadasta/test/resource_tests/test_location_resources.py (hunk 2 above)
Comment (seav) [inline cadasta/test/resource_tests/test_location_resources.py]: Yes. These tests correspond to the tests that were commented out in the original **cadasta_test_suites.py** manual list of tests. I assumed that they were commented out because they were not yet working properly. So I renamed the methods so that pytest doesn't auto-discover and run them. I will eventually go over and check why these tests were commented out and try to debug them.

File: cadasta/test/runtests.py (hunk 3 above)
Comment (seav) [inline cadasta/test/runtests.py]: For some reason, this environment variable is automatically set when you log on to the dev VM. And since we have pytest-django installed, it checks if this environment variable is set and uses that specified Django settings to run pytest under. Otherwise it will just run the plain vanilla pytest.

If you have pytest-django installed and you run pytest in a directory where it can't find the Django settings, pytest fails. Since I am running pytest independent of Django, I have to unset this environment variable, if it is set, in order for the tests to run at all in the current shell session.
